cubi_tk/snappy/kickoff.py

```diff
@@ -12,46 +12,63 @@
 from cubi_tk.exceptions import ParseOutputException
 
 from . import common
+from .snappy_workflows import SnappyWorkflowManager
+
+# from .snappy_workflows import get_snappy_step_directories, get_workflow_step_dependencies
 
 
 def run(
     args, _parser: argparse.ArgumentParser, _subparser: argparse.ArgumentParser
 ) -> typing.Optional[int]:
     logger.info("Try to find SNAPPY pipeline directory...")
     try:
-        path = common.find_snappy_root_dir(args.path or os.getcwd(), common.DEPENDENCIES.keys())
+        path = common.find_snappy_root_dir(args.path or os.getcwd())
     except common.CouldNotFindPipelineRoot:
         return 1
 
-    # TODO: this assumes standard naming which is a limitation...
-    logger.info("Looking for pipeline directories (assuming standard naming)...")
+    logger.info("Looking for pipeline directories (needs to contain snappy config.yaml)...")
     logger.debug("Looking in %s", path)
-    step_set = {name for name in common.DEPENDENCIES if (path / name).exists()}
+
+    manager = SnappyWorkflowManager.from_snappy()
+
+    if manager is None:
+        return None
+
+    step_dependencies = {}
+    folder_steps = manager.get_snappy_step_directories(path)
+    for step_name, step_path in folder_steps.items():
+        dependencies = manager.get_workflow_step_dependencies(step_path)
+        assert all(dep in folder_steps for dep in dependencies)
```

Comment:
It might better to raise a custom error here, since an AssertionError will not clearly communicate that there are missing dependecies in the snappy steps (could happen if i.e. the pipeline gets a major update and one returns to an old project).